src/Interpreters/ExpressionActions.cpp

```diff
@@ -715,6 +729,7 @@ void ExpressionActions::execute(Block & block, size_t & num_rows, bool dry_run,
     {
         .inputs = block.data,
         .num_rows = num_rows,
+        .expression = *this,
```

Comment:
Do we really need this backreference? It's used only in a message for a logical error, so it probably isn't worth introducing because someone later could overuse or misuse it.

Reply:
This is for exception message, could be removed